Userland/Libraries/LibJS/Runtime/TypedArray.h

```diff
@@ -185,19 +185,16 @@ class TypedArray : public TypedArrayBase {
         // 2. Assert: O is an Integer-Indexed exotic object.
 
         // NOTE: If the property name is a number type (An implementation-defined optimized
-        // property key type), it can be treated as a string property that has already been
-        // converted successfully into a canonical numeric index.
+        // property key type), it can be treated as a string property that will transparently be
+        // converted into a canonical numeric index.
 
         // 3. If Type(P) is String, then
         // NOTE: This includes an implementation-defined optimization, see note above!
         if (property_name.is_string() || property_name.is_number()) {
             // a. Let numericIndex be ! CanonicalNumericIndexString(P).
             // NOTE: This includes an implementation-defined optimization, see note above!
```

Comment:
same here (applies everywhere)

Reply:
i agree the bottom NOTE can be removed, but id argue that the top NOTE should stay, because it does deviate from the specification `3. If Type(P) is String, then`:
```c++
if (property_name.is_string() || property_name.is_number()) {
```
Which i would see a suspicous without a comment.

Reply:
agreed, top note should stay, bottom note can go. :+1: